src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java

```diff
@@ -73,7 +73,13 @@ public InteractionImpl(JDAImpl jda, DataObject data)
                 );
             }
             this.channel = channel;
-            user = channel.getUser();
+
+            User user = channel.getUser();
+            if (user == null)
+            {
+                user = jda.getEntityBuilder().createUser(data.getObject("user"));
```

Comment:
Make sure to also set the user on the channel here

Reply:
Good catch, thank you.